From 49815024636c6ca7cc6ebe45b42d7ae33db20127 Mon Sep 17 00:00:00 2001
From: Jakub Filak <jfilak@redhat.com>
Date: Fri, 18 Jul 2014 12:20:06 +0200
Subject: [LIBREPORT PATCH 46/93] gui: apply configuration dialogues changes on
 Enter key

Related to rhbz#1067123

Signed-off-by: Jakub Filak <jfilak@redhat.com>

Conflicts:
	src/gtk-helpers/event_config_dialog.c
	src/gtk-helpers/workflow_config_dialog.c
---
 src/gtk-helpers/event_config_dialog.c    | 4 ++++
 src/gtk-helpers/workflow_config_dialog.c | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/src/gtk-helpers/event_config_dialog.c b/src/gtk-helpers/event_config_dialog.c
index 9d442d6..659f563 100644
--- a/src/gtk-helpers/event_config_dialog.c
+++ b/src/gtk-helpers/event_config_dialog.c
@@ -122,6 +122,7 @@ static void add_option_to_table(gpointer data, gpointer user_data)
                              /*left,top:*/ 0, last_row,
                              /*width,height:*/ 1, 1);
             option_input = gtk_entry_new();
+            gtk_entry_set_activates_default(GTK_ENTRY(option_input), TRUE);
             gtk_widget_set_hexpand(option_input, TRUE);
             if (option->eo_value != NULL)
                 gtk_entry_set_text(GTK_ENTRY(option_input), option->eo_value);
@@ -323,6 +324,9 @@ config_dialog_t *create_event_config_dialog(const char *event_name, GtkWindow *p
                         GTK_RESPONSE_APPLY,
                         NULL);
 
+
+    gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_APPLY);
+
     /* Allow resize?
      * W/o resize, e.g. upload configuration hint looks awfully
      * line wrapped.
diff --git a/src/gtk-helpers/workflow_config_dialog.c b/src/gtk-helpers/workflow_config_dialog.c
index 45d7fb6..7b7fe32 100644
--- a/src/gtk-helpers/workflow_config_dialog.c
+++ b/src/gtk-helpers/workflow_config_dialog.c
@@ -89,6 +89,8 @@ config_dialog_t *create_workflow_config_dialog(const char *workflow_name, GtkWin
                         GTK_RESPONSE_APPLY,
                         NULL);
 
+    gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_APPLY);
+
     gtk_window_set_resizable(GTK_WINDOW(dialog), true);
     gtk_window_set_default_size(GTK_WINDOW(dialog), 450, 450);
 
-- 
1.8.3.1

